website/static/js/filepage/index.js

```diff
@@ -209,9 +209,9 @@ var FileViewPage = {
                 }, ctrl.editor.title);
             }
         };
-
         m.render(document.getElementById('toggleBar'), m('.btn-toolbar.m-t-md', [
-            ctrl.canEdit() ? m('.btn-group.m-l-xs.m-t-xs', [
+            // Special case whether or not to show the delete button for published Dataverse files
+            (ctrl.canEdit() && $(document).context.URL.indexOf('version=latest-published') < 0 ) ? m('.btn-group.m-l-xs.m-t-xs', [
```

Comment:
Discussed use of .context with @mfraezz . Though it's deprecated as of JQ 1.10, not using it lead to a race condition (document.url and window.location.href didn't work sans JQ solution).

We'll go with this for now, but keep a note in case it's an issue later.